src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -41,39 +41,35 @@ public static bool HasPdb(string module)
         public static void CopyCoverletDependency(string module)
         {
             var directory = Path.GetDirectoryName(module);
-            if (Path.GetFileNameWithoutExtension(module) == "coverlet.core")
-                return;
+            var moduleFileName = Path.GetFileName(module);
 
             var assembly = typeof(Coverage).Assembly;
             string name = Path.GetFileName(assembly.Location);
+            if (name == moduleFileName)
+                return;
+
             File.Copy(assembly.Location, Path.Combine(directory, name), true);
         }
 
         public static void BackupOriginalModule(string module, string identifier)
         {
-            var backupPath = Path.Combine(
-                Path.GetTempPath(),
-                Path.GetFileNameWithoutExtension(module) + "_" + identifier + ".dll"
-            );
-
+            var backupPath = GetBackupPath(module, identifier);
             File.Copy(module, backupPath);
         }
 
         public static void RestoreOriginalModule(string module, string identifier)
         {
-            var backupPath = Path.Combine(
-                Path.GetTempPath(),
-                Path.GetFileNameWithoutExtension(module) + "_" + identifier + ".dll"
-            );
+            var backupPath = GetBackupPath(module, identifier);
 
             // Restore the original module - retry up to 10 times, since the destination file could be locked
             // See: https://github.com/tonerdo/coverlet/issues/25
             var currentSleep = 6;
-            Func<TimeSpan> retryStrategy = () => {
+            TimeSpan retryStrategy()
```

Comment:
Since the retry strategy is the same in `InstrumentationHelper.cs`, we can just create a single method and use that everywhere